server/gameplay/game_hander_test.go

```diff
@@ -0,0 +1,250 @@
+package gameplay
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	d "github.com/MettyS/checkers/server/domain"
+)
+
+func playersFixture() map[string]d.GameRole {
+	return map[string]d.GameRole{
+		"player1": d.PlayerWhite,
+		"player2": d.PlayerBlack,
+		"player3": d.Spectator,
+	}
+}
+
+func gameWithPlayersFixture() Game {
+	game := Game{}
+	game.Board = []d.Tile{2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}
+	game.CurrentPlayer = d.PlayerWhite
+	game.Participants = playersFixture()
+	return game
+}
+
+func gameWithBlackTurn() Game {
+	game := Game{}
+	game.Board = []d.Tile{2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1}
+	game.CurrentPlayer = d.PlayerBlack
+	game.Participants = playersFixture()
+	return game
+}
+
+func (g Game) String() string {
+	return fmt.Sprintf("Board: %v\nCurrent player: %v \nParticipants: %v\n", g.Board, g.CurrentPlayer, g.Participants)
+}
+
+func getTileString(tile d.Tile) string {
+	if tile == d.White {
+		return "ww"
+	} else if tile == d.Black {
+		return "bb"
+	} else if tile == d.WhitePromoted {
+		return "pw"
+	} else if tile == d.BlackPromoted {
+		return "pb"
+	} else {
+		return "ee"
+	}
+}
+
+type (
+	addParticipantTest struct {
+		testname     string
+		player       string
+		expectedRole d.GameRole
+		expectedErr  error
+	}
+)
+
+func TestGameSetup(t *testing.T) {
+	game := CreateGameHandler()
+	fmt.Println(boardString(game))
+	var addPlayerTests = []addParticipantTest{
+		{"Add unique playerID1",
+			"player1", d.PlayerWhite, nil},
+		{"Add unique playerID2", "player2", d.PlayerBlack, nil},
+		{"Add unique playerID3", "player3", d.Spectator, nil},
+		{"Don't add repeat playerID1", "player1", d.PlayerWhite,
+			d.Error{Message: "Player already exists (shouldn't happen with random tokens)"},
+		},
+		{"Add unique playerID4", "player4", d.Spectator, nil},
+	}
+
+	for _, addPlayerTest := range addPlayerTests {
+		testname := addPlayerTest.testname
+		t.Run(testname, func(t *testing.T) {
+			resultRole, resultError := game.AddParticipant(addPlayerTest.player)
+
+			if addPlayerTest.expectedRole != resultRole || addPlayerTest.expectedErr != resultError {
+				t.Errorf("\nFailure - \nRecieved Role: %v , Error: %v\nExpected : Role: %v , Error: %v",
+					resultRole, resultError, addPlayerTest.expectedRole, addPlayerTest.expectedErr)
+			}
+		})
+	}
+	fmt.Println(game.String())
+}
+
+func buildMoveSet(nums []int) []d.Move {
+	moves := make([]d.Move, 0)
+	var indexFrom, indexTo uint32 = uint32(math.MaxUint32), uint32(math.MaxUint32)
+	for i, num := range nums {
+		if i%2 == 0 {
+			indexFrom = uint32(num)
+		} else {
+			indexTo = uint32(num)
+			moves = append(moves, d.Move{IndexFrom: indexFrom, IndexTo: indexTo})
+		}
+	}
+	return moves
+}
+
+type (
```

Comment:
Can you change this to just `type name whatever` without the multiple lines and parens?

Reply:
Thanks for catching this.